src/analysis/special_functions/trigonometric/angle.lean

```diff
@@ -427,6 +476,18 @@ begin
       abs_eq_self.2 h.1]
 end
 
+lemma abs_to_real_eq_pi_div_two_iff {θ : angle} :
+  |θ.to_real| = π / 2 ↔ (θ = (π / 2 : ℝ) ∨ θ = (-π / 2 : ℝ)) :=
```

Comment:
Do you have lemmas for `θ.to_real = π / 2 ↔ ...` and `θ.to_real = -(π / 2) ↔ ...`? If yes, then you can use `abs_eq`, then these lemmas.

Reply:
I've now added those lemmas and revised this proof accordingly. (I think it's still useful to have this lemma itself, not just the others, since "angle is a right angle" is something that may appear in several different forms in other arguments and so lemmas converting between those different forms are useful.)